js/src/jsmath.cpp

```diff
@@ -259,6 +319,27 @@ bool js::math_imul_handle(JSContext* cx, HandleValue lhs, HandleValue rhs,
     return false;
   }
 
+  // TaintFox
+  NumberObject *taintedResult = NumberObject::create(cx, 0);
+  bool isTainted = false;
+
+  if(isTaintedNumber(lhs)){
```

Comment:
Another thing I noticed - I think a lot of this code can be replaced by a call to JS::getAnyNumberTaint in jstaint.cpp.

Have a look at e.g. the AddOperation in Interpreter-inl.h for an example how this is done.

I have recently (last week!) updated the logic for combining two taint flows together so that in principle both parents in the taint flow are saved. If you use the getAnyNumberTaint call then all of this will be taken in account automatically.

Reply:
PS - you might need to update your branch to the HEAD of primitaint-merge to get the latest changes.